File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import pojo.TbBrand;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findBrands(){
        return brandService.findAll();
    }

     @RequestMapping("/findPage")
    public PageResult findPage(int pageNum, int pageSize){
        return brandService.findPage(pageNum,pageSize);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){
        try {
            brandService.add(brand);
            return new Result(true, "增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "增加失败");
        }
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }
    @RequestMapping("/delete")
    public Result delete(Long []ids){
        try {
            brandService.delete(ids);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand, int page, int size  ){
        return brandService.findPage(brand, page, size);
    }
    @RequestMapping("/selectBrandOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectBrandOptionList();
    }

}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
//定义 品牌控制器  brandController  自定义服务也是依赖注入 按照名称注入
app.controller("brandController",function($scope,$controller,brandService){

    //继承 写法 伪继承
    $controller("baseController",{$scope:$scope})
    //查询所有的方法
    $scope.findAll=function(){
        brandService.findAll().success(function(response){
            $scope.list=response
        })
    }


    $scope.findPage=function(pageNum,pageSize){
        brandService.findPage(pageNum,pageSize).success(function(response){
            /**
             * 总页数total :
             每页记录rows :[]
             */
            $scope.list= response.rows  //当前页记录
            $scope.paginationConf.totalItems=response.total  //赋值总页数
        })
    }

    $scope.save=function(){
        var objectReust= null
        //根据条件()判断是增加还是修改
        if($scope.entity.id==null){
            //增加
            objectReust = brandService.add($scope.entity)

        }else{
            //修改
            objectReust= brandService.update($scope.entity)

        }

        objectReust.success(function(response){
            if(response.success){
                // alert(response.message)
                $scope.reloadList() //刷新列表
            }else{
                alert(response.message)
            }

        })


    }

    $scope.findOne=function(id){
        brandService.findOne(id).success(function (response) {
            $scope.entity=response // TbBrand 转成json 就是response
        })
    }

    $scope.del=function(){
        brandService.del($scope.selectIds).success(function(response){
            if(response.success){
                // alert(response.message)
                $scope.reloadList() //刷新列表
                $scope.selectIds=[];
            }else{
                alert(response.message)
            }
        })
    }



    $scope.searchEntity = {} //空对象没有值

    $scope.search=function(page,size){
        brandService.search(page,size,$scope.searchEntity).success(function(response){
            /**
             * 总页数total :
             每页记录rows :[]
             */
            $scope.list= response.rows  //当前页记录
            $scope.paginationConf.totalItems=response.total  //赋值总页数
        })
    }


})